src/lib.rs

```diff
@@ -80,22 +114,20 @@ impl Ale {
 	}
 
 	/// Resets the Atari and loads a game from the file specified.
-	/// 
+	///
 	/// After this call the game should be ready to play. This is necessary after changing a
 	/// setting for the setting to take effect.
 	pub fn load_rom_file(&mut self, rom_file: &CStr) {
-		unsafe { ale_sys::loadROM(self.ptr, rom_file.as_ptr()); }
+		unsafe {
+			ale_sys::loadROM(self.ptr, rom_file.as_ptr());
+		}
 	}
 
 	/// Applies an action to the game and returns the reward.
-	/// 
+	///
 	/// It is the user's responsibility to check if the game has ended and reset
 	/// when necessary - this method will keep pressing buttons on the game over screen.
-	///
-	/// # Panics
-	/// If the action is not legal.
 	pub fn act(&mut self, action: i32) -> i32 {
-		assert!(self.legal_action_set().contains(&action), "Illegal action: {}", action);
```

Comment:
Why was this removed? Is it needed for your use-case?

Reply:
Hey, sorry for the late reply, missed the notification for this. I'm also not sure about the failing CI, it seems to be unrelated to this PR.

I removed this to avoid copying the legal_action_set vector before every step and to match the C++ ALEInterface where illegal actions are converted into noops here, see here:
https://github.com/mgbellemare/Arcade-Learning-Environment/blob/master/src/environment/stella_environment.cpp#L210

Reply:
Sounds good to me, thanks :) I'll merge in and publish a new version